doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,396 @@
+```
+will result in five files in a directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+```
+This would result in five datums being processed in any pipelines that use this repo.
+
+Now, take a one-line file containing
+```
+Six
+```
+and load it into Pachyderm appending it to the count.txt file.  If that file were named, "more-count.txt", the command might look like
+```
+pachctl put-file line-data master my-data.txt -f more-count.txt --split line
+```
+That will result in six files in the directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+count.txt/0000000000000004: Six
+```
+This would result in one datum being processed in any pipelines that use this repo: the new file `count.txt/0000000000000004`.
+
+#### Overwriting files with --split
+The behavior of Pachyderm when a file loaded with `--split` is overwritten is simple to explain but subtle in its implications.  Remember that the loaded file will be split into those sequentially-named files, as shown above.  If any of those resulting split-files hashes differently than the one it's replacing, that will cause the Pachyderm Pipeline System to process that data.
+
+This can have important consequences for downstream processing.  For example, let's say you have that same file named "count.txt" consisting of 5 lines that we used in the previous example
+```
+One
+Two
+Three
+Four
+Five
+```
+As discussed prior, loading that file into Pachyderm using `--split` will result in five files in a directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+```
+This would result in five datums being processed in any pipelines that use this repo.
+
+Now, modify that file by inserting the word "Zero" on the first line.
+```
+Zero
+One
+Two
+Three
+Four
+Five
+```
+Let's upload it to Pachyderm using `--split` and `--overwrite`.
+```
+pachctl put-file line-data master count.txt -f ./count.txt --split line --overwrite
+```
+The input repo will now look like this
+```
+count.txt/0000000000000000: Zero
+count.txt/0000000000000001: One
+count.txt/0000000000000002: Two
+count.txt/0000000000000003: Three
+count.txt/0000000000000004: Four
+count.txt/0000000000000005: Five
+``` 
+As far as Pachyderm is concerned, every single file existing has changed, and a new file has been added.  Six datums would be processed by a downstream pipelines.
```

Comment:
I'd mention explicitly here that the reason for this is that the filename is taken into account when hashing the data for the pipeline. This will only result in 1 new piece of content being stored "Zero", but all of them will get reprocessed.